### PR TITLE
PM-13432: Update toast design to support title and subtitle

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-13432](https://bitwarden.atlassian.net/browse/PM-13432)

## 📔 Objective

Updates the design of the toasts to match the new design. This new design includes support for a title and subtitle which required a bit of refactoring from just the text that existed previously.

- 447e79b30e9b7034a985c700a1662fb427e6e5ce Refactors existing toast instances to use title instead of text
- f00a7e82f8ed1242806d6d75adc287ab4e54165c Updates the design of the toast

## 📸 Screenshots

Screenshot 2024-10-11 at 4 19 33 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-13432]: https://bitwarden.atlassian.net/browse/PM-13432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Platform/Application/Views/ToastView.swift [hunk 1]
```diff
@@ -27,28 +49,42 @@ struct ToastView: View {
...
+            .clipShape(RoundedRectangle(cornerRadius: 8))
+            .shadow(color: .black.opacity(0.1), radius: 8, x: 0, y: 2)
+            .shadow(color: .black.opacity(0.05), radius: 4, x: 0, y: 4)
+            .accessibilityElement(children: .combine)
+            .padding(.horizontal, 16)
+            .task(id: toast.id) {
+                do {
+                    try await Task.sleep(nanoseconds: 3 * NSEC_PER_SEC)
+                    withAnimation {
+                        self.toast = nil
```
Comment (phil-livefront) [inline BitwardenShared/UI/Platform/Application/Views/ToastView.swift:82]: this is not really possible to test, right?

File: BitwardenShared/UI/Platform/Application/Utilities/Coordinator.swift [hunk 2]
```diff
@@ -172,10 +174,12 @@ extension Coordinator where Self: HasNavigator {
...
     ///
-    /// - Parameter text: The text of the toast to display.
+    /// - Parameters:
+    ///   - title: The title text displayed in the toast.
+    ///   - title: The subtitle text displayed in the toast.
     ///
-    func showToast(_ text: String) {
-        navigator?.showToast(Toast(text: text))
+    func showToast(_ title: String, subtitle: String? = nil) {
+        navigator?.showToast(Toast(title: title, subtitle: subtitle))
```
Comment (phil-livefront) [inline BitwardenShared/UI/Platform/Application/Utilities/Coordinator.swift:182]: same question here, we can't test this, right?

**Review by phil-livefront — DISMISSED** [review]
Looks great!

File: BitwardenShared/UI/Platform/Application/Views/ToastView.swift (hunk 1 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Views/ToastView.swift:82]: Right, I don't think there's a way we can trigger this with view inspector.

File: BitwardenShared/UI/Platform/Application/Utilities/Coordinator.swift (hunk 2 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Application/Utilities/Coordinator.swift:182]: Added a test for this!

**Review by phil-livefront — APPROVED** [review]

**Pull request merged**